src/main/java/Duke.java

```diff
@@ -1,10 +1,95 @@
+import java.util.Scanner;
+
 public class Duke {
+    static final String BREAK = System.lineSeparator();
+    static final String TAB = "    ";
+    static final String BORDER = "    ____________________________________________________________";
+
+    private static TaskManager taskManager;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
+        showLogo();
+        greet();
+        interact();
+        sayBye();
+    }
+
+    private static void interact() {
+        // create input scanner
+        Scanner in = new Scanner(System.in);
+
+        taskManager = new TaskManager();
+        while (in.hasNextLine()) {
+            String userInputLine = in.nextLine();
+
+            if (inputIsBye(userInputLine)) {
+                break;
+            } else if (inputIsList(userInputLine)) {
+                listTasks("Here are the tasks in your list:");
+            } else if (inputIsDone(userInputLine) && taskIsChecked(userInputLine)) {
+                listTasks("Nice! I've marked this task as done:");
+            } else {
+                Task taskAdded = taskManager.addTask(userInputLine);
+                showAddTask(taskAdded);
+            }
+        }
+
+        in.close();
+    }
+
+    private static void showAddTask(Task task) {
+        String prologue = TAB + "Got it. I've added this task:" + BREAK;
+        String content = TAB + "  " + task.toString() + BREAK;
+        String epilogue = TAB + "Now you have " + taskManager.getTotalTasksNumber() + " tasks in the list." + BREAK;
+        printBetwBorders(prologue + content + epilogue);
+    }
+
+    private static boolean taskIsChecked(String userInputLine) {
+        return taskManager.taskIsChecked(userInputLine);
+    }
```

Comment:
Should this be called directly as taskManager.taskIsChecked(userInputLine) in the main function?